### PR TITLE
Automated tests of the dashboard

### PR DESCRIPTION
**Pull request by marek-siemieniuk-morawski**
## What does this PR do?

- Automated tests of the dashboard, displaying transaction details and filtering transactions
- Extracted methods like `passThroughOnboarding` to external module `steps` from page objects
- Extracted page object of dashboard screen to external module  

#### *Any scout cleaning/refactoring?*

- Modified `TransactionStatus` enum 

## Todos:

- [ ] Checked on iOS
- [ ] Checked on Android

## Required reviewers:
@DawidBastek

### CODE REVIEW

File: tests/e2e/pageObjects/types.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export type StandardWalletType = '3-Key Vault' | '2-Key Vault' | 'Standard HD P2SH';
+
+export type WalletType = StandardWalletType | 'Standard P2SH' | 'Standard HD SegWit';
```
Comment (DawidBastek) [inline tests/e2e/pageObjects/types.ts:3]: (?) What do you think to change `StandardWalletType` to i.e. `BasicWalletType` or `DefaultWalletType` or something else? `Standard could bring some misunderstanding in names where you have 3 types of Standard wallets.

File: tests/e2e/pageObjects/types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export type StandardWalletType = '3-Key Vault' | '2-Key Vault' | 'Standard HD P2SH';
```
Comment (DawidBastek) [inline tests/e2e/pageObjects/types.ts:1]: (?) What do you think to change the file name from `types.js` to something more descriptive i.e. `walletTypes.js`?

Comment (marek-siemieniuk-morawski) [reply tests/e2e/pageObjects/types.ts]: Not really. The file includes 3 other types, it's not only about the wallet types.

Apart from that, there's not a convention in TS to have a file per single type/interface. The TypeScript team uses as well a file called `types.ts`: https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts

File: tests/e2e/pageObjects/types.ts (hunk 1 above)
Comment (marek-siemieniuk-morawski) [reply tests/e2e/pageObjects/types.ts]: Good point. Done

File: tests/e2e/pageObjects/pages/FiltersScreen.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  async chooseContactToFilterBy(contactName: string) {
+    await actions.tap(this.filterByFromSenderInput);
+  },
+
+  async selectFromDateFilter() {
+    // No idea how to implement this with the current date picker component
+    throw new Error('Not implemented');
+  },
+
+  async selectToDateFilter() {
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply tests/e2e/pageObjects/pages/FiltersScreen.ts:43]: As above

File: tests/e2e/pageObjects/pages/FiltersScreen.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    await actions.tap(this.filterBySentOption);
+  },
+
+  async chooseContactToFilterBy(contactName: string) {
+    await actions.tap(this.filterByFromSenderInput);
+  },
+
+  async selectFromDateFilter() {
+    // No idea how to implement this with the current date picker component
+    throw new Error('Not implemented');
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply tests/e2e/pageObjects/pages/FiltersScreen.ts:40]: As above

File: tests/e2e/pageObjects/pages/FiltersScreen.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+  async tapOnSentFilterOption() {
+    await actions.tap(this.filterBySentOption);
+  },
+
+  async chooseContactToFilterBy(contactName: string) {
+    await actions.tap(this.filterByFromSenderInput);
+  },
+
+  async selectFromDateFilter() {
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply tests/e2e/pageObjects/pages/FiltersScreen.ts:38]: Well, not really but I want to leave a comment with info on why it's not done.

File: tests/e2e/pageObjects/pages/Dashboard.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    walletsCarousel: element(by.id('wallet-carousel')),
+    walletsDropdown: element(by.id('wallets-dropdown')),
+    getWalletDropdownItem: (walletName: string) => element(by.id(`wallet-dropdown-${walletName}-item`)),
+    getWalletCardElement: (walletName: string) => element(by.id(`wallet-${walletName}-card`)),
+    getWalletDetailsButtonElement: (walletName: string) => element(by.id(`show-${walletName}-details-button`)),
+
+    noWalletsIcon: element(by.id('no-wallets-icon')),
+
+    recieveButton: element(by.id('receive-coins-button')),
+    recoverButton: element(by.id('recover-coins-button')),
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply tests/e2e/pageObjects/pages/Dashboard.ts]: Good point, fixed.

File: tests/e2e/actions.ts [hunk 7]
```diff
@@ -49,18 +49,49 @@ const Actions = () => {
...
+  const swipeCarousel = async (carousel: Detox.DetoxAny, direction: 'left' | 'right') => {
+    await waitFor(carousel)
+      .toBeVisible()
+      .withTimeout(WAIT_FOR_ELEMENT_TIMEOUT);
+
+    await carousel.swipe(direction, 'fast', 0.75, 0.5);
+  };
+
+  /** Scrolls down and up to find an element */
+  const searchForElement = async (target: Detox.DetoxAny, scrollable: string) => {
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply tests/e2e/actions.ts:70]: 🙂

File: src/screens/Dashboard/WalletsCarousel.tsx [hunk 8]
```diff
@@ -34,6 +34,7 @@ export class WalletsCarousel extends Component<Props> {
     return (
       <View>
         <Carousel
+          testID="wallet-carousel"
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply src/screens/Dashboard/WalletsCarousel.tsx]: Fixed

File: src/screens/Dashboard/DashboardContentHeader.tsx [hunk 9]
```diff
@@ -40,6 +40,7 @@ export const DashboardContentHeader = ({
   return (
     <View style={styles.header}>
       <Dropdown
+        testID="wallets-dropdown"
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply src/screens/Dashboard/DashboardContentHeader.tsx]: Fixed

File: tests/e2e/pageObjects/index.ts [hunk 10]
```diff
@@ -21,6 +24,9 @@ const app = {
   betaVersionScreen: BetaVersionScreen(),
   developerRoom: DeveloperRoom(),
   onboarding: Onboarding(),
+  dashboard: Dashboard(),
+  filterScreen: FiltersScreen(),
```
[the remark this answers is not in the document]
Comment (marek-siemieniuk-morawski) [reply tests/e2e/pageObjects/index.ts]: Not at all :) Fixed

**Review by DawidBastek — APPROVED** [review]

**Pull request merged**